src/parquet/bloom_filter_algorithm.cc

```diff
@@ -16,11 +16,40 @@
 // under the License.
 
 #include "parquet/bloom_filter_algorithm.h"
+#include "arrow/status.h"
+#include "arrow/util/bit-util.h"
+#include "parquet/exception.h"
 #include "parquet/types.h"
 
 namespace parquet {
 constexpr uint32_t BlockBasedAlgorithm::SALT[BITS_SET_PER_BLOCK];
 
+BlockBasedAlgorithm::BlockBasedAlgorithm(uint32_t num_bytes)
+    : pool_(::arrow::default_memory_pool()) {
+  if ((num_bytes & (num_bytes - 1)) != 0) {
+    throw ParquetException("num_bytes is not a power of 2");
+  }
+
+  num_bytes_ = num_bytes;
+
+  ::arrow::Status status =
+      pool_->Allocate(num_bytes, reinterpret_cast<uint8_t**>(&bitset32_));
```

Comment:
This comment from last time was not addressed, I think:

Incorrect alignment can produce incorrect compilation even without SIMD. If we treat the result like a `uint32_t *` and it is not 4-byte aligned, then the compilation can produce surprising and incorrect code.

The documentation says that Allocate returns 8-byte aligned results. Please add a check that it is at least `alignof(decltype(bitset_[0]))`.